Gems/Atom/RHI/Code/Source/RHI.Reflect/ConstantsLayout.cpp

```diff
@@ -149,5 +147,20 @@ namespace AZ
 
             return true;
         }
+
+        void ConstantsLayout::DebugPrintNames(AZStd::array_view<ShaderInputConstantIndex> constantList) const
+        {
+            AZStd::string output;
+            for (const ShaderInputConstantIndex& constandIdx : constantList)
+            {
+                if (constandIdx.GetIndex() < m_inputs.size())
+                {
+                    output += m_inputs[constandIdx.GetIndex()].m_name.GetCStr();
```

Comment:
Can this information be accessed from outside the ConstantsLayout? I'm just wondering if this function can be moved to your ShaderResourceGroupDebug files. Following proper OO design of single responsibility, ConstantsLayout shouldn't be doing string building or debug printing.

Reply:
m_inputs is not directly accessible outside the class, which means we can't test
constandIdx.GetIndex() < m_inputs.size()